pkg/provisioner/controller.go

```diff
@@ -319,10 +319,13 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	}
 
 	updateOnSuccess := func() error {
```

Comment:
The source of the problem really is embedding the function and allowing it to modify variables in the outer context. This is such a long function anyway, the preferred pattern is to define the function separately and return the computed value(s). Then upon success the caller can assign the variables in its context. But refactoring this would be better saved for another day so we can get the critical fix in today.